src/components/StaticWebsite.ts

```diff
@@ -77,7 +79,26 @@ export class StaticWebsite extends CdkConstruct implements Construct {
             comment: `Identity that represents CloudFront for the ${id} static website.`,
         });
 
+        /**
+         * CloudFront function that manipulates the HTTP responses to add security headers.
+         */
+        const code = `function handler(event) {
+    var response = event.response;
+    response.headers = Object.assign({}, {
+        'x-frame-options': {value: 'SAMEORIGIN'},
+        'x-content-type-options': {value: 'nosniff'},
+        'x-xss-protection': {value: '1; mode=block'},
+        'strict-transport-security': {value: 'max-age=63072000'},
```

Comment:
Here are the security headers being added to the response.

Reply:
Do we want to consider extending this with `; includeSubDomains; preload`? It adds more risk (if someone is unaware, they can cause all subdomains to be required to use HTTPS) but also makes a production application more secure (as browsers can then force the domain to HTTPS internally even if a user has never visited it). Maybe a mode setting?

Reply:
That's a good point, but I would be cautious about adding that right now. Indeed, here we control the static website, but users could host anything in subdomains, so I wouldn't want to break that.

Adding an option is probably the best course of action. In that case, I'd focus on covering the main use case first and add the option later.